stack.c

```diff
@@ -46,6 +49,32 @@ int mill_max_cached_stacks = 64;
 static int mill_num_cached_stacks = 0;
 static struct mill_slist mill_cached_stacks = {0};
 
+static void *mill_allocstackmem(size_t stack_size) {
+#if MILL_ENABLE_STACK_GUARD
+    long pagesize = sysconf(_SC_PAGE_SIZE);
+    if (!(pagesize > 0)) {
+        return NULL;
+    }
+    mill_assert(mill_stack_size > pagesize);
+
+    void *ptr;
+    int rc = posix_memalign(&ptr, pagesize, mill_stack_size);
+    if (rc != 0) {
+        errno = rc;
+        return NULL;
+    }
+
+    rc = mprotect(ptr, pagesize, PROT_NONE);
+    if (rc != 0) {
```

Comment:
Yes, thats on purpose. From `man 3 posix_memalign`

    posix_memalign() returns zero on success, or one of the error values listed
    in the next section on failure.  The value of errno is indeterminate after
    a call to posix_memalign().